client/utils/qrcode.js

```diff
@@ -0,0 +1,55 @@
+export function qrCodeStringToObject(str) {
+  let [id, name, category, imageUrl, price, unit] = str.split(',');
+  return {
+    id,
+    name,
+    category,
+    imageUrl,
+    unit,
+    price: parseFloat(price)
+  }
+}
+
+export function fileToImageBuffer(file) {
+  return new Promise((resolve /*, reject*/) => {
+    var imageType = /^image\//;
+    if (!imageType.test(file.type)) {
+      throw new Error('File type not valid');
+    }
+    // Read file
+    var reader = new FileReader();
+    reader.addEventListener('load', () => {
+      let img = new Image();
+      let canvas = document.createElement('canvas');
+      img.onload = () => {
+        canvas.width = img.width;
+        canvas.height = img.height;
+        let ctx = canvas.getContext('2d');
+        ctx.drawImage(img,0,0);
+        let data = ctx.getImageData(0, 0, img.width, img.height);
+        resolve(data);
+      }
+      img.src = event.target.result;
+    }, false);
+
+
+    reader.readAsDataURL(file);
+  });
+}
+
+export function onQrCodeScan(imageBuffer, cartStore) {
+  // WEB WORKER SOLUTION
```

Comment:
This function and the contents of `client/qr-worker.js` is where students will be writing their code for this exercise. Nothing else needs to be touched in order to move from "Sync qr code analysis" (results in a 3-10 second freeze in the app) to "worker qr code analysis"